tests/testthat/test-Rules-methods.R

```diff
@@ -1453,7 +1465,55 @@ test_that("StoppingSpecificDose correclty replaces next best string with specifi
   )
   expected <- structure(
     TRUE,
-    message = "12 patients lie within 5% of the specific dose 80. This reached the required 9 patients"
+    message = "12 patients lie within 5% of the specific dose 80. This reached the required 9 patients",
+    report_label = character(0)
   )
   expect_identical(result, expected)
 })
+
+test_that("Default label assignment for stopping rules works correctly.", {
+  my_data <- h_get_data()
+  my_model <- h_get_logistic_kadane()
+  my_samples <- mcmc(my_data, my_model, h_get_mcmc_options(samples = 1000, burnin = 1000))
+
+  my_stopping <- StoppingMinCohorts(nCohorts = 3, report_label = "")
+
+  result <- stopTrial(
+    stopping = my_stopping,
+    model = my_model,
+    samples = my_samples,
+    dose = 7,
+    data = my_data
+  )
+
+  expect_equal(attr(result, "report_label"), "Minimum number of 3 cohorts reached")
+})
+
+
+test_that("Customized label assignment for stopping rules works correctly.", {
+  my_data <- h_get_data()
+  my_model <- h_get_logistic_kadane()
+  my_samples <- mcmc(my_data, my_model, h_get_mcmc_options(samples = 1000, burnin = 1000))
+
+  my_stopping <- StoppingMinCohorts(nCohorts = 3, report_label = "a customized label")
+
+
+  result <- stopTrial(
+    stopping = my_stopping,
+    model = my_model,
+    samples = my_samples,
+    dose = 7,
+    data = my_data
+  )
+
+  result <- stopTrial(
+    stopping = my_stopping,
+    model = my_model,
+    samples = my_samples,
+    dose = 7,
+    data = my_data
+  )
+
+
+  expect_equal(attr(result, "report_label"), "a customized label")
+})
```

Comment:
Ok, but we need to test it for each stopping class, not only for `StoppingMinCohorts`. So I would propose that we always test for a default label (within the existing unit tests) + we need a separate short test for custom label for each class.

Reply:
this is still not solved.